Source/Model/SPNarrowDownCompletion.m

```diff
@@ -785,18 +788,15 @@ - (void)watchUserEvents
 
 	[theView setCompletionIsOpen:YES];
 
-	closeMe = NO;
 	while(!closeMe) {
 		NSEvent* event = [NSApp nextEventMatchingMask:NSEventMaskAny
 		                                    untilDate:[NSDate distantFuture]
 		                                       inMode:NSDefaultRunLoopMode
 		                                      dequeue:YES];
+    startedIntercepting = YES;
 
 		if(!event) continue;
 
-		// Exit if closeMe has been set in the meantime
-		if(closeMe) return;
```

Comment:
How does this `closeMe` loop ends? We are in a `while` loop right?

Reply:
There are several `break;` statements within the while loop and there are also several methods calls such as `[self filter];` and `[self close]` and `[self completeAndInsertSnippet];` that sets `closeMe = YES`.

My reasoning for removing this check here is that we have intercepted an event already and it was `dequeue` so it seems that if we return here that event will goes unprocessed / is lost. By continuing we should at least call `[NSApp sendEvent:event];` so that it continues to propagate through the first-responder hierarchy and on the next iteration the `while(!closeMe)` condition will stop the looping.